Jenkinsfile

```diff
@@ -1,67 +1,145 @@
-                stage('Build and test gcc-7 container') { steps { buildAndTest('7', 'yask') } }
+                stage('Build and test gcc-4.9 container') {
+                     agent { dockerfile { label 'azure-linux'
+                                          filename 'Dockerfile.jenkins'
+                                          additionalBuildArgs "--build-arg gccvers=4.9" } }
+                     environment { 
+                         HOME="${WORKSPACE}"
+                         DEVITO_OPENMP=0
+                         PYTHONPATH="${WORKSPACE}/lib/python3.6/site-packages/"
+                     }
+                     steps {
+                         cleanWorkspace()
+                         pipInstallDevito()
+                         runPipTests()
+                     }
+                }
+                stage('Build and test gcc-4.9 OpenMP container') {
+                     agent { dockerfile { label 'azure-linux' 
+                                          filename 'Dockerfile.jenkins'
+                                          additionalBuildArgs "--build-arg gccvers=4.9" } }
+                     environment { 
+                         HOME="${WORKSPACE}"
+                         DEVITO_OPENMP=1
+                         OMP_NUM_THREADS=2
+                     }
+                     steps {
+                         cleanWorkspace()
+                         condaInstallDevito()
+                         runCondaTests()
+                         runExamples()
+                     }
+                }
+                stage('Build and test gcc-5 container') {
+                     agent { dockerfile { label 'azure-linux' 
+                                          filename 'Dockerfile.jenkins'
+                                          additionalBuildArgs "--build-arg gccvers=5" } }
+                     environment { 
+                         HOME="${WORKSPACE}"
+                         DEVITO_OPENMP=0
+                     }
+                     steps {
+                         cleanWorkspace()
+                         condaInstallDevito()
+                         runCondaTests()
+                         runExamples()
+                     }
+                }
+                stage('Build and test gcc-7 container') {
+                     agent { dockerfile { label 'azure-linux'
+                                          filename 'Dockerfile.jenkins'
+                                          additionalBuildArgs "--build-arg gccvers=7" } }
+                     environment { 
+                         HOME="${WORKSPACE}"
+                         DEVITO_BACKEND="yask"
+                         DEVITO_OPENMP="0"
+                         YC_CXX="g++-7"
+                     }
+                     steps {
+                         cleanWorkspace()
+                         condaInstallDevito()
+                         installYask()
```

Comment:
I have this feeling that with Jenkins we could just have a separate YASK build ?

Reply:
Could you elaborate on that a bit? As in, would you like to have a gcc7 build without yask and then another one with yask? That's very easy indeed.

Reply:
no sorry, my bad, I was confused by something else :)